equinox/nn/pool.py

```diff
@@ -85,6 +91,26 @@ def __init__(
                 f"{num_spatial_dims} containing ints or tuples of length 2."
             )
 
+        for padding, kernel_size in zip(self.padding, self.kernel_size):
+            if kernel_size / 2 < padding[0] or kernel_size / 2 < padding[1]:
+                raise ValueError(
+                    "Padding should be less than or equal to half of the kernel size."
+                    f" Received kernel_size {kernel_size} and padding {padding}."
```

Comment:
I don't think this check is technically necessary. If someone really wanted they might want to add loads of padding, knowing they'd just get zero out.

Reply:
I see. With a case where `padding == kernel`, I was getting garbage values at the start and sometimes at the end. Do you think raising a warning here would be useful ? 
```python
x = jnp.arange(14).reshape(1, 14)
max_pool = eqx.nn.MaxPool1d(3, 3, padding=3)
print(max_pool(x))

#[[-2147483648 , 2 , 5 , 8, 11, 13]]
```

Reply:
So the "garbage values" are the identity value for the operation in question. At a conceptual level it makes sense that they're there.

On a practical level though, I can believe that this would probably surprise most folks, and may be a silent source of bugs for them. So I'm happy to restrict padding if you prefer.

Reply:
If we're keeping this check, should it actually be `if kernel_size > left_padding + right_padding`?

Reply:
Or actually perhaps `if kernel_size >= max(left_padding, right_padding)`?

Since as long as the padding on each side is smaller than the window, eaxh window must necessarily include at least one non-padding value.

Reply:
> `kernel_size >= max(left_padding, right_padding)`

This seems like a reasonable check.